Lib/idlelib/editor.py

```diff
@@ -672,14 +672,11 @@ def replace_event(self, event):
 
     def goto_line_event(self, event):
         text = self.text
-        lineno = tkSimpleDialog.askinteger("Goto",
-                "Go to line number:",parent=text)
-        if lineno is None:
-            return "break"
-        if lineno <= 0:
-            text.bell()
-            return "break"
-
+        lineno = query.Goto(
```

Comment:
Does this behave correctly when simply closing the dialog?

Reply:
No. Cursor stays put but selection is erroneously cleared.  Fixed.